src/main/java/hudson/plugins/sshslaves/verifiers/TrileadVersionSupportManager.java

```diff
@@ -0,0 +1,70 @@
+package hudson.plugins.sshslaves.verifiers;
+
+import com.trilead.ssh2.signature.DSASHA1Verify;
+import com.trilead.ssh2.signature.RSASHA1Verify;
+import hudson.plugins.sshslaves.Messages;
+
+import java.io.IOException;
+
+/**
+ * @author Michael Clarke
+ * @since 1.18
+ */
+final class TrileadVersionSupportManager {
+
+    static TrileadVersionSupport getTrileadSupport() {
+        try {
+            Thread.currentThread().getContextClassLoader().loadClass("com.trilead.ssh2.signature.KeyAlgorithmManager");
+            return createaVersion9Instance();
+        } catch (ReflectiveOperationException e) {
+            // KeyAlgorithmManager doesn't exist, fall back to legacy trilead handler
+            return new LegacyTrileadVersionSupport();
+        }
+    }
+
+    private static TrileadVersionSupport createaVersion9Instance() {
+        try {
+            return (TrileadVersionSupport) Class.forName("hudson.plugins.sshslaves.verifiers.JenkinsTrilead9VersionSupport").newInstance();
+        } catch (ReflectiveOperationException e) {
```

Comment:
ReflectiveOperationException covers ClassNotFoundException as well as the InvocationException that would be thrown if the class could not be instantiated